### PR TITLE
Resistivities

### PR DESCRIPTION
**Pull request by rochSmets**
include resistivity and hyper_resistivity as coming from the dict filled via Python in `populateDict`

### CODE REVIEW

File: src/solver/level_initializer/hybrid_level_initializer.h [hunk 1]
```diff
@@ -35,7 +35,9 @@ namespace solver
         static constexpr auto interp_order = GridLayoutT::interp_order;
 
 
-        PHARE::core::Ohm<GridLayoutT> ohm_;
+        PHARE::core::Ohm<GridLayoutT>
+            ohm_(dict); // where the fuck should this dict come from ? should we make an explicit
```
Comment (PhilipDeegan) [inline src/solver/level_initializer/hybrid_level_initializer.h:39]: either 

```
explicit HybridLevelInitializer(PHARE::core::Ohm<GridLayoutT>&& ohm) : ohm_{ohm} {
```

or

```
explicit HybridLevelInitializer(PHARE::initializer::PHAREDict dict) : ohm_{dict} {
```

I suppose, the former is minimal, the latter is more extensible should "ampere_" need parameters later for example

File: pyphare/pyphare/pharein/simulation.py [hunk 2]
```diff
@@ -536,6 +560,9 @@ def __init__(self, **kwargs):
           self.nSubcycles ** levelNumbers[ilvl] * self.time_step_nbr for ilvl in levelNumbers
         ]
 
+        print("resistivity : ", self.resistivity)
```
Comment (nicolasaunai) [inline pyphare/pyphare/pharein/simulation.py:563]: probably not needed now?

File: src/core/numerics/ohm/ohm.h [hunk 3]
```diff
@@ -10,14 +10,29 @@
...
     template<typename GridLayout>
     class Ohm : public LayoutHolder<GridLayout>
     {
+    public:
+        explicit Ohm(PHARE::initializer::PHAREDict dict)
+            : eta_{dict["resistivity"].template to<double>()}
+            , nu_{dict["hyper_resistivity"].template to<double>()}
+        {
+        }
+        // Ohm() = default;
```
Comment (nicolasaunai) [inline src/core/numerics/ohm/ohm.h:30]: remove?

File: src/solver/level_initializer/level_initializer_factory.h [hunk 4]
```diff
@@ -17,11 +18,12 @@ namespace solver
...
 
     public:
-        static std::unique_ptr<LevelInitializer<AMRTypes>> create(std::string modelName)
+        static std::unique_ptr<LevelInitializer<AMRTypes>>
+        create(std::string modelName, PHARE::initializer::PHAREDict dict)
         {
             if (modelName == "HybridModel")
             {
-                return std::make_unique<HybridLevelInitializer<HybridModel>>();
+                return std::make_unique<HybridLevelInitializer<HybridModel>>(dict);
```
Comment (nicolasaunai) [inline src/solver/level_initializer/level_initializer_factory.h:26]: should we do std::move(dict) here?

File: tests/core/numerics/ohm/test_main.cpp [hunk 5]
```diff
@@ -303,19 +320,34 @@ TYPED_TEST_SUITE(OhmTest, OhmTupleInfos);
 
 
 
-TYPED_TEST(OhmTest, ThatOhmHasCtor)
+TYPED_TEST(OhmTest, ThatOhmHasDefaultCtor) // this test should be deprecated in a close future as
```
Comment (nicolasaunai) [inline tests/core/numerics/ohm/test_main.cpp:323]: is it still needed?

File: tests/functional/alfven_wave/alfven_wave1d.py [hunk 6]
```diff
@@ -21,15 +21,17 @@ def config():
...
+        smallest_patch_size = 20,
+        largest_patch_size = 20,
+        time_step_nbr = 2,           # number of time steps (not specified if time_step and final_time provided)
+        time_step = 0.001,           # simulation time step (not specified if final_time and time_step_nbr provided)
+        boundary_types = "periodic", # boundary condition, string or tuple, length == len(cell) == len(dl)
+        cells = 40,                  # integer or tuple length == dimension
+        dl = 0.2,                    # mesh size of the root level, float or tuple
+        refinement_boxes = {"L0": {"B0": [(10, ), (20, )]}},
+        diag_options = {"format": "phareh5", "options": {"dir": "alfven_norefine","mode":"overwrite"}},
+        resistivity = 0.01,
```
Comment (nicolasaunai) [inline tests/functional/alfven_wave/alfven_wave1d.py:33]: I'm not sure why the need to modify the alfven wave test in that PR?

File: src/core/numerics/ohm/ohm.h [hunk 7]
```diff
@@ -239,7 +254,7 @@ namespace core
         template<typename VecField, typename ComponentTag>
         auto resistive_(VecField const& J, MeshIndex<VecField::dimension> index, ComponentTag) const
         {
-            auto const eta = 1.0; // TODO : eta should comme from input file
+            auto const eta = this->eta_;
```
Comment (PhilipDeegan) [inline src/core/numerics/ohm/ohm.h:257]: Possibly unnecessary copy?

File: src/solver/level_initializer/level_initializer_factory.h (hunk 4 above)
Comment (PhilipDeegan) [reply src/solver/level_initializer/level_initializer_factory.h:26]: > should we do std::move(dict) here?

I think it's not needed as it's RVO, a copy shoud(would?) cause a compile error

Comment (PhilipDeegan) [reply src/solver/level_initializer/level_initializer_factory.h:26]: I think it's not needed as it's RVO, a copy shoud(would?) cause a compile error

Comment (PhilipDeegan) [reply src/solver/level_initializer/level_initializer_factory.h:26]: http://coliru.stacked-crooked.com/a/c5f1ed9aec4a50b4

File: src/core/numerics/ohm/ohm.h (hunk 3 above)
Comment (rochSmets) [reply src/core/numerics/ohm/ohm.h:30]: not clear to why should we remove these 4 lines... init `nu_` is important no ?

File: src/solver/level_initializer/level_initializer_factory.h (hunk 4 above)
Comment (rochSmets) [reply src/solver/level_initializer/level_initializer_factory.h:26]: so it should stays as is ?

File: src/core/numerics/ohm/ohm.h (hunk 7 above)
Comment (rochSmets) [reply src/core/numerics/ohm/ohm.h:257]: just because `eta` is shorter than `this->eta_`

Comment (PhilipDeegan) [reply src/core/numerics/ohm/ohm.h:257]: sorry, I meant the lack of ref(&)

```
auto const& eta = this->eta_;
```

I think copying doubles is super fast anyway so maybe it's not all that important.

File: src/solver/level_initializer/level_initializer_factory.h (hunk 4 above)
Comment (nicolasaunai) [reply src/solver/level_initializer/level_initializer_factory.h:26]: I meant 
```
 return std::make_unique<HybridLevelInitializer<HybridModel>>(std::move(dict)); 
```

Comment (PhilipDeegan) [reply src/solver/level_initializer/level_initializer_factory.h]: ohh right, no I don't think it's needed
maybe just make this[1] a const ref, I added a few const functions to dict.hpp so it should be ok

[1] https://github.com/rochSmets/PHARE/blob/resistivities/src/solver/level_initializer/hybrid_level_initializer.h#L47

```
explicit HybridLevelInitializer(PHARE::initializer::PHAREDict const& dict)
```

if const fails, let me know, we should assess that I think

Comment (rochSmets) [reply src/solver/level_initializer/level_initializer_factory.h]: done

**Review by nicolasaunai — APPROVED** [review]

**Review by PhilipDeegan — APPROVED** [review]

**Pull request merged**